crates/hir/src/display.rs

```diff
@@ -251,22 +222,44 @@ impl HirDisplay for Union {
         write!(f, "{}", self.name(f.db).display(f.db.upcast()))?;
         let def_id = GenericDefId::AdtId(AdtId::UnionId(self.id));
         write_generic_params(def_id, f)?;
+
         let has_where_clause = write_where_clause(def_id, f)?;
+        if let Some(limit) = f.entity_limit {
+            display_fields_or_variants(&self.fields(f.db), has_where_clause, limit, f)?;
+        }
+        Ok(())
+    }
+}
 
-        let fields = self.fields(f.db);
-        if !fields.is_empty() {
-            f.write_char(if !has_where_clause { ' ' } else { '\n' })?;
-            f.write_str("{\n")?;
-            for field in self.fields(f.db) {
-                f.write_str("    ")?;
-                field.hir_fmt(f)?;
-                f.write_str(",\n")?;
-            }
-            f.write_str("}")?;
+fn display_fields_or_variants<T: HirDisplay>(
```

Comment:
I think we should split this into two, one for fields one  for variants. Reason being for an enum like
```rs
enum Enum {
    Variant {},
    Variant2 { field: Field },
    Variant3(),
    Variant4(Field),
    Variant5,
}
```
hovering the enum should yield
```rs
enum Enum {
    Variant {},
    Variant2 { .. }, // or the field if the limit is enabled and non-zero
    Variant3(),
    Variant4(,.), , // or the field if the limit is enabled and non-zero
    Variant5,
}
```
That is hovering an enum should only show its shape, not its vairants fields. To see those the person can hover the variant instead.

Reply:
> Variant2 { .. }, // or the field if the limit is enabled and non-zero

Here I use a single config to limit the num of fields or variants of ADTs, should we separate it into two different ones?

Reply:
AH wait I made a mistake, hovering enums should ***not*** render the variants fields.

Splitting the two into two configs sounds reasonable nevertheless.

Reply:
So, for hovering on enums, we can display it as 🤔

```rust
enum Enum {
    Variant {},
    Variant2 { /* .. */ },
    Variant3(),
    Variant4( /* .. */ ),
    Variant5,
}
```